Search Improvements

SaraVieira opened the pull request:
In a user test it was pointed we can't order the search and it makes sense to do so

This PR adds order by likes and views


This PR also:

* Changes the cards in search for the new cards we have in the explore page
* Removes how long the search took as that is not important information for the user
* Changes the API of wide sandbox to have the prop `selectSandbox` instead of `pickSandbox` to make it more generic
* Adds common script
* Moves some .css files to gobal js styles so we can use the theme

Design
https://www.figma.com/file/akq4c3BMhkW5EHipeIBEbKmf/Search-Page?node-id=0%3A1

SaraVieira commented:
cc @Haroenv

Haroenv commented:
The sorting of "relevance" and "views" seems very similar to me, was a different sort required for this? 

I'm seeing some flashes when I change index, but I don't see any particular reason why in the code by a quick glance, maybe it already was there before

![2019-03-21 15-53-19 2019-03-21 15_54_37](https://user-images.githubusercontent.com/6270048/54761121-c8d34000-4bf1-11e9-89d6-eafb9dc12192.gif)

Also, it seems like the search is pretty hidden with the new design, I can't seem to find any link towards it anymore, I had to guess the URL.

Finally there's a small other thing (maybe only on Safari), and that's the search icon isn't correctly positioned: 

<img width="1325" alt="Screenshot 2019-03-21 at 15 56 36" src="https://user-images.githubusercontent.com/6270048/54761202-f3bd9400-4bf1-11e9-85b1-a09cd95b930d.png">

SaraVieira commented:
Good points!

I deleted the views one as that is basically the default one and also added a link to the search in the header

The other two:

The flickering seems to only happen on safari :/

I'm looking at the CSS thing now

CompuIves commented:
The bottom part of the card (with description) seems a bit big to me, does it have different styles compared to `SandboxCard` in `common`?

SaraVieira commented:
@CompuIves Nop, itmay be because most of them don't have a description

CompuIves commented on packages/app/src/app/pages/Search/Results/index.js:
```diff
@@ -2,17 +2,30 @@ import React from 'react';
-      <Hits hitComponent={SandboxCard} />
+      <Hits
+        hitComponent={({ hit }) => (
+          <WideSandbox
```
It does seem a bit weird, after checking it seems like the description of the sandbox in search is ~150px and in explore ~130px height

CompuIves commented on packages/common/src/components/WideSandbox/index.tsx:
```diff
@@ -34,7 +34,7 @@ export type Props = {
   defaultHeight?: number;
   noMargin?: boolean;
-  pickSandbox: (
+  selectSandbox: (
```
Nice!!

CompuIves commented on packages/app/src/app/pages/Search/Filters/index.js:
```diff
@@ -1,12 +1,22 @@
 function Filters() {
   return (
     <Container>
+      <Sort
```
This is nice! Maybe we can also re-use 

![image](https://user-images.githubusercontent.com/587016/54824409-1c559480-4cab-11e9-80cd-bc641446e2c4.png)

in a later version.

CompuIves commented on package.json:
```diff
@@ -75,7 +76,8 @@
     "lint-staged": "^5.0.0",
-    "prettier": "^1.8.2"
+    "prettier": "^1.8.2",
+    "typescript": "^3.3.4000"
```
Could this one be `^3.3.1`? That way we ensure that we have a single typescript version in the whole repo.

SaraVieira replied:
Fixed
